src/Analyser/NodeScopeResolver.php

```diff
@@ -1664,6 +1664,9 @@ private function findEarlyTerminatingExpr(Expr $expr, Scope $scope): ?Expr
 		}
 
 		$exprType = $scope->getType($expr);
+		if ($expr instanceof Expr\Match_ && $exprType instanceof NeverType) {
+			return $expr;
```

Comment:
I'm not sure about this fix. What should be done instead is that these situations should be caught by the `if ($exprType instanceof NeverType && $exprType->isExplicit()) {` instead.

"Implicit" NeverType is for example when you try to intersect two incompatible types (`string&int`). "Explicit" mixed is when you have a `throw` expression, function with `@return never` and similar.

This means that the explicitness of "NeverType" gets lost when we have a match expression with "explicit never" arms, and that's something that should be fixed instead.

Reply:
Thank you for the review and informative examples. I'll try fixing in that way!